### PR TITLE
Add proto_toolchain rule

### PR DESCRIPTION
**Pull request by Yannic**
Progress on https://docs.google.com/document/d/1go3UMwm2nM8JHhI3MZrtyoFEy3BYg5omGqQmOwcjmNE/edit

### CODE REVIEW

**Comment by Yannic** [comment]
@katre PTAL

/cc @comius

File: tools/proto/private/rules/proto_toolchain.bzl [hunk 1]
```diff
@@ -0,0 +1,48 @@
+"""A toolchain for Protobuf rules."""
+
+load("//tools/config:common_settings.bzl", "BuildSettingInfo")
+
+def _proto_toolchain_impl(ctx):
+    compiler = ctx.attr.compiler
+    compiler_options = ctx.attr.compiler_options
+
+    # TODO(yannic): Migrate native `--protocopt` to Starlark flag when
+    # accumulating multiple flags to a list works correctly.
```
[the remark this answers is not in the document]
Comment (Yannic) [reply tools/proto/private/rules/proto_toolchain.bzl:10]: Filed one yesterday: https://github.com/bazelbuild/bazel/issues/13817

File: tools/proto/private/rules/proto_toolchain.bzl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    command_line_compiler_options = getattr(ctx.fragments.proto, "protoc_opts", [])
+
+    return [
+        platform_common.ToolchainInfo(
+            proto = ProtoToolchainInfo(
```
[the remark this answers is not in the document]
Comment (Yannic) [reply tools/proto/private/rules/proto_toolchain.bzl:15]: It's in the global namespace

**Comment by lberki** [comment]
In that case, I'll bounce this to @comius , who is closest to protocol buffers than I am these days

**Comment by Yannic** [comment]
ping?

**Pull request closed without merging**